src/tempo.cpp

```diff
@@ -127,4 +127,27 @@ int Tempo::ResetDrawing(FunctorParams *functorParams)
     return FUNCTOR_CONTINUE;
 }
 
+int Tempo::CalcMaxMeasureDuration(FunctorParams *functorParams)
+{
+    CalcMaxMeasureDurationParams *params = vrv_params_cast<CalcMaxMeasureDurationParams *>(functorParams);
+    assert(params);
+
+    if (this->HasMidiBpm()) {
+        params->m_currentTempo = this->GetMidiBpm();
+    }
+    else if (this->HasMm()) {
+        double mm = this->GetMm();
+        int mmUnit = 4;
+        if (this->HasMmUnit() && (this->GetMmUnit() > DURATION_breve)) {
+            mmUnit = pow(2, (int)this->GetMmUnit() - 2);
+        }
+        if (this->HasMmDots()) {
+            mmUnit = 2 * mmUnit - (mmUnit / pow(2, this->GetMmDots()));
+        }
+        params->m_currentTempo = mm * 4.0 / mmUnit + 0.5;
+    }
+
+    return FUNCTOR_CONTINUE;
+}
+
```

Comment:
If I remember correctly the reason why we did not do that is because putting it there will make the tempo change be parsed after the content of the measure, because the data structure is typically:
```xml
<measure>
   <staff/>
   <staff/>
   <tempo/>
</measure>
```
This is why we had a tempo `lookup` in Measure. Could you double check? Thanks

Reply:
Yay! 👍

Reply:
Yes that's why I introduced an end functor. Now the content of the measure (including tempo changes and multi measure rests) is evaluated and after that all the updates (`m_currentTempo`, `m_currentScoreTime`, `m_currentRealTimeSeconds`) are set in `Measure::CalcMaxMeasureDurationEnd`.

If we enhance the example in the issue with
```xml
<tempo midi.bpm="60" staff="1" tstamp="1.000000"></tempo>
```
in the second measure (the one with the multi measure rest), then the time map becomes:
```json
[
	{
		"tstamp":	0.000000,
		"qstamp":	0.000000,
		"tempo":	120.000000,
		"on":	["n61e36o"]
	},
	{
		"tstamp":	2000.000000,
		"qstamp":	4.000000,
		"off":	["n61e36o"]
	},
	{
		"tstamp":	14000.000000,
		"qstamp":	16.000000,
		"tempo":	60.000000,
		"on":	["n6cte68"]
	},
	{
		"tstamp":	18000.000000,
		"qstamp":	20.000000,
		"off":	["n6cte68"]
	}
]
```
This looks good to me, since now we spend 4*3 seconds on the multi measure rest.

Reply:
Ah yes, I overlooked that change. Excellent, thanks!